_build/sharpmake/src/base.cs

```diff
@@ -53,14 +53,26 @@ private void SetupDefaultConfigurationSettings(RexConfiguration conf, RexTarget
     conf.Options.Add(Options.Vc.General.CharacterSet.MultiByte);
     conf.Options.Add(Options.Vc.General.PlatformToolset.v142);
     conf.Options.Add(Options.Vc.General.WarningLevel.Level4);
-    conf.Options.Add(Options.Vc.General.TreatWarningsAsErrors.Enable);
+
+    // Disable warning as errors in debug so that we can add debug code
+    // without the compiler warning about it.
+    // We just need to make sure we delete it before it goes in to version control
```

Comment:
INFO: this will be captured by the CI when it's still present. CI will also run Release and this will trigger the warning as error.

Thanks for implementing the feature request.

Reply:
true, the warning as error will be triggered, but ideally you want this at submit time..
we can perform a compilation of the files we've changed only on when you submit them to capture these issues.
you really want to capture these things as early as possible. another way of doing it is by manually tagging these lines and let the submit step parse a file looking for tags which would block submission

Reply:
I would not complicate things with extra tags
You can trigger a release compilation before you commit, but this will prevent us to push code if it does not compile ( which is maybe what you want ) but it will slow down development if you would like to push code from one place to another or give somebody else control of the code if it does not compile.